### PR TITLE
 [BUGFIX] JSON Serialize RowCondition and MetricBundle computation result to enable IDDict.to_id() for SparkDFExecutionEngine

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
Whenever an entity that is not yet serialized undergoes `IDDict().to_id()` operation, the `JSON` operation in the latter encounters an exception.  Hence, these `compute_domain_kwargs` need to be `JSON`-serialized first.  The problem does not manifest itself when `compute_domain_kwargs` contains only a single key (due to the way `IDDict` works).  Hence, one should always add a second key (e.g., `batch_id`) in order to test for such a circumstance.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1216
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/execution_engine/test_sqlalchemy_execution_engine.py [hunk 1]
```diff
@@ -790,6 +791,74 @@ def test_resolve_metric_bundle_with_nonexistent_metric(sa):
...
+            {
+                "names": [
+                    "Ada Lovelace",
+                    "Alan Kay",
+                    "Donald Knuth",
+                    "Edsger Dijkstra",
+                    "Guido van Rossum",
+                    "John McCarthy",
+                    "Marvin Minsky",
+                    "Ray Ozzie",
```
Comment (Shinnnyshinshin) [inline tests/execution_engine/test_sqlalchemy_execution_engine.py:809]: ❤️

File: tests/execution_engine/test_sqlalchemy_execution_engine.py [hunk 2]
```diff
@@ -790,6 +791,74 @@ def test_resolve_metric_bundle_with_nonexistent_metric(sa):
         print(e)
 
 
+def test_resolve_metric_bundle_with_compute_domain_kwargs_json_serialization(sa):
+    """
+    Insures that even when "compute_domain_kwargs" has multiple keys, it will be JSON-serialized for "IDDict.to_id()".
+    """
```
Comment (Shinnnyshinshin) [inline tests/execution_engine/test_sqlalchemy_execution_engine.py:797]: you even did an `sa` one! :) thank you

**Review by Shinnnyshinshin — APPROVED** [review]
Love it. Love it. Thank you @alexsherstinsky

File: tests/execution_engine/test_sqlalchemy_execution_engine.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/execution_engine/test_sqlalchemy_execution_engine.py:797]: @Shinnnyshinshin It had to be both -- they are subject to the same situation (`RowCondition` which begins as not yet serialized).  Thanks!

**Pull request merged**